server/s3_object_metadata.h

```diff
@@ -347,5 +373,95 @@ class S3ObjectMetadata : private S3ObjectMetadataCopyable {
   FRIEND_TEST(S3ObjectMetadataTest, GetEncodedBucketAcl);
 };
 
+// Fragment or the part detail structure
+struct s3_part_frag_context {
+  struct m0_uint128 motr_OID;
+  struct m0_uint128 PVID;
+  struct m0_uint128 versionID;
+  size_t item_size;
+  int layout_id;
+  bool is_multipart;
+};
+
+// TBD
+// Class to read/write S3 object parts and fragments from object list index
+// e.g., see object list table entry below:
+// ObjectOne                  <Basic object metadata> + versionID, FNo, PRTS​
+// ObjectTwo                  <Basic object metadata> + versionID, FNo, PRTS
+// ~ObjectOne|versionID|F1    OID1, PVID, size, layout-id1​
+// ~ObjectTwo|versionID|P1    OID2, PVID, size, layout-id​2
+// ~ObjectTwo|versionID|P2|F1 OID3, PVID, size, layout-id​3
+// ~ObjectTwo|versionID|P2|F2 OID4, PVID, size, layout-id​4
+class S3ObjectExtededMetadata : private S3ObjectMetadataCopyable {
+ private:
+  std::string bucket_name;
```

Comment:
whether all these members needed here,or something derived from base class can be used

Reply:
All members are needed. They are not in base class 'S3ObjectMetadataCopyable'